### PR TITLE
Update dependency on PyTorch >=2.0.1

### PR DESCRIPTION
**Pull request by mzweilin**
# What does this PR do?

This PR updates dependency on PyTorch to >=2.0.1 for better compatibility.

## Type of change

Please check all relevant options.

- [x] Improvement (non-breaking)
- [ ] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Testing

Please describe the tests that you ran to verify your changes. Consider listing any relevant details of your test configuration.

- [x] `pytest`
- [x] `CUDA_VISIBLE_DEVICES=0 python -m mart experiment=CIFAR10_CNN_Adv trainer=gpu trainer.precision=16` reports 70% (21 sec/epoch).
- [ ] `CUDA_VISIBLE_DEVICES=0,1 python -m mart experiment=CIFAR10_CNN_Adv trainer=ddp trainer.precision=16 trainer.devices=2 model.optimizer.lr=0.2 trainer.max_steps=2925 datamodule.ims_per_batch=256 datamodule.world_size=2` reports 70% (14 sec/epoch).

## Before submitting

- [x] The title is **self-explanatory** and the description **concisely** explains the PR
- [x] My **PR does only one thing**, instead of bundling different changes together
- [ ] I list all the **breaking changes** introduced by this pull request
- [ ] I have commented my code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have run pre-commit hooks with `pre-commit run -a` command without errors

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by dxoigmn — APPROVED** [review]
LGTM! Although you might consider target torch >= 2.0. Similar with torchvision.

File: pyproject.toml [hunk 1]
```diff
@@ -11,8 +11,8 @@ authors = [
 requires-python = ">=3.9"
 
 dependencies = [
-  "torch ~= 2.0.1",
-  "torchvision ~= 0.15.2",
+  "torch >= 2.0",
+  "torchvision >= 0.15",
```
Comment (dxoigmn) [inline pyproject.toml:15]: I would make this:
```
torch >= 2.0.1
torchvision >= 0.15.2
```

**Review by dxoigmn — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 1 above)
Comment (dxoigmn) [reply pyproject.toml:15]: Actually that make not work correctly. Let me lookup the syntax again.

Comment (dxoigmn) [reply pyproject.toml:15]: Actually, maybe that does work? I think it's the `~` and `^` specifiers that affect major/minor/patch updating. See: https://python-poetry.org/docs/dependency-specification/

Comment (mzweilin) [reply pyproject.toml:15]: Yes, `>=2.0.1` should work. 

However, we don't use poetry in MART.

Comment (mzweilin) [reply pyproject.toml]: Fixed in 7b852d493db616ef81ca8cf4a459a5abf9b06b18

**Review by dxoigmn — APPROVED** [review]

**Pull request merged**